libp2p/peer/peerstore.py

```diff
@@ -10,90 +12,196 @@
 
 class PeerStore(IPeerStore):
 
-    peer_map: Dict[ID, PeerData]
+    peer_pubkey_map: Dict[ID, PublicKey]
+    peer_privkey_map: Dict[ID, PrivateKey]
+    peer_data_map: Dict[ID, PeerData]
 
     def __init__(self) -> None:
-        IPeerStore.__init__(self)
-        self.peer_map = {}
+        self.peer_pubkey_map = {}
+        self.peer_privkey_map = {}
+        self.peer_data_map = {}
 
     def __create_or_get_peer(self, peer_id: ID) -> PeerData:
         """
         Returns the peer data for peer_id or creates a new peer data (and
-        stores it in peer_map) if peer data for peer_id does not yet exist.
+        stores it in peer_data_map) if peer data for peer_id does not yet
+        exist.
 
         :param peer_id: peer ID
         :return: peer data
         """
-        if peer_id in self.peer_map:
-            return self.peer_map[peer_id]
+        if peer_id in self.peer_data_map:
+            return self.peer_data_map[peer_id]
         data = PeerData()
-        self.peer_map[peer_id] = data
-        return self.peer_map[peer_id]
+        self.peer_data_map[peer_id] = data
+        return self.peer_data_map[peer_id]
```

Comment:
this is super nit-picky but i expect the bytecode is marginally more efficient if we just return `data` rather than re-accessing the map. i would actually hope a "sufficiently smart compiler" would just do this for us but i don't know well enough to say